### PR TITLE
Debugging and Providing Stability

### PR DESCRIPTION
**Pull request by smusali**
Versions:
* `Node.js`: `v8.11.3`
* `NPM`: `5.6.0`

The following features have successfully tested and fixed:
* `register [email]`
* `ssologin`
* `login [email]`
* `tail [options] [query]`
* `switch`
* `search [options] [query]`
* `info|whoami`
* `update`

### CODE REVIEW

**Comment by smusali** [comment]
@dchai76, can you review this whenever you have time? Thanks in advance!

File: lib/config.js [hunk 1]
```diff
@@ -1,14 +1,20 @@
 module.exports = {
-    VERSION_CHECK_URL: 'https://repo.logdna.com/PLATFORM/version'
-    , UPDATE_CHECK_URL: 'https://repo.logdna.com/PLATFORM/logdna.gz'
-    , UPDATE_CHECK_INTERVAL: 86400000 // 1 day
-    , SSO_URL: 'https://logdna.com/sso/'
-    , SSO_LONG_PATH: '/ext/cli-sso/'
-    , SSO_POLL_INTERVAL: 5000 // 5s
-    , DEFAULT_CONF_FILE: '~/.logdna.conf'.replace('~', process.env.HOME || process.env.USERPROFILE)
+    DEFAULT_CONF_FILE: '~/.logdna.conf'.replace('~', process.env.HOME || process.env.USERPROFILE)
+    , DEFAULT_TIMEOUT: 2500
```
Comment (dchai76) [inline lib/config.js]: You think you should comment on the value of this like you do with `SSO_POLL_INTERVAL` and elsewhere? i.e. `// 2.5s`

File: package.json [hunk 2]
```diff
@@ -20,8 +20,7 @@
   },
   "homepage": "https://github.com/logdna/logdna-cli",
   "dependencies": {
-    "bluebird": "^3.5.3",
-    "bufferutil": "^1.3.0",
+    "bufferutil": "^4.0.1",
```
Comment (dchai76) [inline package.json]: Not this line, but I can't seem to comment on a line that's not edited - we have a high security issue for `ws` - can you bump up that version?

File: lib/utils.js [hunk 3]
```diff
@@ -118,32 +136,35 @@ const apiPost = function(config, endpoint, params, callback) {
 
 const renderLine = function(config, line, params) {
 
+    if (!line._host || !line._app) return;
```
Comment (dchai76) [inline lib/utils.js]: Error msg?

File: lib/utils.js [hunk 4]
```diff
@@ -6,17 +7,19 @@ const request = require('request');
...
 const ERROR_LEVEL_TEST = /[Ee]rr(?:or)?|ERR(?:OR)?|[Cc]rit(?:ical)?|CRIT(?:ICAL)?|[Ff]atal|FATAL|[Ss]evere|SEVERE|EMERG(?:ENCY)?|[Ee]merg(?:ency)?/;
 const WARN_LEVEL_TEST = /[Ww]arn(?:ing)?|WARN(?:ING)?/;
 
 const saveConfig = function(config, callback) {
     properties.stringify(config, {
         path: config.DEFAULT_CONF_FILE
     }, function(err) {
-        if (err) return callback('Error while saving to: ' + (config.DEFAULT_CONF_FILE) + ': ' + err);
+        if (err) {
+            return callback(`Error in Saving ${JSON.stringify(config)} into: ${config.DEFAULT_CONF_FILE}: ${err}`);
```
Comment (dchai76) [inline lib/utils.js]: One observation - all of this code is this strange unholy mix of styles - `var`s mixed with `let/const`s, string concat style (`'Err in saving' + val`) mixed with interpolation (`\`Err in saving ${val}\``). It makes for a jarring read - would prefer we be consistent and convert throughout.

**Comment by smusali** [comment]
Well, @dchai76, I think, I have already removed `smusali:logdna-cli` repo before waiting for getting this reviewed/merged - so, closing this for now.

**Pull request closed without merging**